AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -560,6 +560,28 @@ - (void)displayImmediately
   [[self asyncLayer] displayImmediately];
 }
 
+- (void)__setNeedsLayout
+{
+  ASDisplayNodeAssertThreadAffinity(self);
+  ASDN::MutexLocker l(_propertyLock);
+  
+  if (!_flags.isMeasured) {
+    return;
+  }
+  
+  ASSizeRange oldConstrainedSize = _constrainedSize;
+  [self invalidateCalculatedLayout];
+  
+  if (_supernode) {
+    // Cause supernode's layout to be invalidated
+    [_supernode setNeedsLayout];
+  } else {
+    // This is the root node. Trigger a full measurement pass on *current* thread. Old constrained size is re-used.
+    [self measureWithSizeRange:oldConstrainedSize];
+    self.frame = CGRectMake(0.0f, 0.0f, _layout.size.width, _layout.size.height);
```

Comment:
Important: my most key questions on this diff relate to this method.

- What impact will it have to not call the view /or/ layer's -setNeedsLayout method?
- I think we must include that call.  If the view or layer has not been created yet, then -setNeedsLayout should be set in the pending state.
- If we did not include this call, it seems like nodes that are visited by the -measure pass may never get the new layout actually applied to them.  Particularly if they are not the root node, and if the -measure pass does not touch anything else that may trigger a CoreAnimation relayout.
- I believe omitting the call may break some existing code, so if it is actually essential that we /not/ call it, share your thinking on why.